vllm/model_executor/layers/quantization/utils/w8a8_utils.py

```diff
@@ -194,13 +194,28 @@ def apply_int8_linear(
     weight: torch.Tensor,
     weight_scale: torch.Tensor,
     input_scale: Optional[torch.Tensor] = None,
+    input_zero_point: Optional[torch.Tensor] = None,
+    azp_adj: Optional[torch.Tensor] = None,
     bias: Optional[torch.Tensor] = None,
 ):
     # ops.scaled_int8_quant supports both dynamic and static quant.
     # * dynamic, layer.input_scale is None and x_scale computed from x.
     # * static, layer.input_scale is scalar and x_scale is input_scale.
-    x_q, x_scale = ops.scaled_int8_quant(input, input_scale)
-
+    symmetric = azp_adj is None
+    x_q, x_scale, x_zp = ops.scaled_int8_quant(input,
+                                               input_scale,
+                                               input_zero_point,
+                                               symmetric=symmetric)
+
+    if x_zp is not None:
+        return ops.cutlass_scaled_mm_azp(x_q,
```

Comment:
Do you think it would be better to have `ops.cutlass_scaled_mm` with optional azp parms instead of branching here?

Reply:
I just didn't want the `cutlass_scaled_mm` to blow up in complexity. Also, in the dynamic case, we don't pass `azp`  because it's calculated, so we'd need an extra parameter to distinguish the symmetric and asymmetric cases